spec/rspec_otel/matchers/emit_span_spec.rb

```diff
@@ -157,6 +157,13 @@
         span.finish
       end.not_to emit_span('test').without_event('testing an event')
     end
+
+    it 'matches a span without an event' do
```

Comment:
In implementing `with_link/without_link` I found that this test would fail, which I don't think it should and was due to putting nil checks in the `event_match?` method. So I pulled them out with the appropriate logic in the `with_event/without_event`